rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,403 @@
+
+This RFC will focus on what approach and strategies we shall take in our Design System.
+
+# Basic Example
+
+### Responsive Modal
+
+👇🏻 An example of a responsive design where we have modal which is responsive to the screen size and has a `maxWidth` defined for it to render on larger screen sizes, where on smaller screen sizes(mobile) it'll take `100%` of the width of the screen.
+
+<img alt="Responsive Modal" src="./images/responsive-adaptive-layout/responsive-modal.png">
+
+<br/>
+
+### Adaptive Search Field
+
+👇🏻 An example of a adaptive design where we have two implementation of a search field.
+
+1. Larger Screen Size - Search Input with Autosuggestion
+2. Smaller Screen Size - Search Field when clicked opens up a full screen modal which basically gives an optimised search experience on smaller screen size.
+
+<img alt="Responsive Modal" src="./images/responsive-adaptive-layout/adaptive-search-field.png">
+
+# Motivation
+
+### Why are we doing this?
+
+As of today, we don't have any strategy or approach in place for designing layouts for different screen sizes. Ensuring that elements fit within a page is not enough. There are different experiences that works for different screen sizes.
+
+Just going Responsive(fluid) design blindly is not correct. Imagine for example, you have different experiences(styles, javascript) for mobile vs desktop but if you just go blindly with responsive we might unnecessary download assets for all the screen sizes our application would run on. For example it doesn't makes sense to implement [this Search Field](#adaptive-search-field) as a responsive layout since it'll result in unnecessary download of the assets(modal for mobile) and hence hampering the experience and performance for the consumers.
+
+Similarly, just going adaptive blindly is not correct too. We might unnecessary be creating redundant variant of the same component. For example, imagine that we are creating 2 variants of a modal - 1 for desktop(which has a width of `600px`) and 1 for mobile(which has `100%` width). Now we could have just made it responsive by setting the `maxWidth` property of the modal so it could have been `600px` in width on desktop and `100%` in width on mobile. For eg, it doesn't makes sense to create multiple variant of [this Modal](#responsive-modal) while we can achieve this through responsive layout by setting `maxWidth` for Modal's Dialog component.
+
+But because we don't have these strategies in place we don't know how to approach building layouts and components for different screen sizes.
+
+### What use cases does it support?
+
+- Handling responsiveness internally in the components based on different screen sizes.
+- Layout components that will help you define different layouts based on screen sizes.
+- Rendering different variants of a component based on screen sizes, optimising for performance without bloating the bundlesize.
```

Comment:
Had to rephrase this a little since "Components for Adaptive layout(rendering components conditionally)" doesn't fit well here
```suggestion
- Components that render different variants conditionally
```